app/questionnaire/rules/utils.py

```diff
@@ -1,6 +1,12 @@
 from datetime import datetime, timezone
 from typing import Optional, overload
 
+ISO_8601_DATETIME_FORMAT = "%Y-%m-%dT%H:%M:%S%z"
+
+
+def parse_iso_8601_datetime(iso_8601_datetime: str) -> datetime:
+    return datetime.fromisoformat(iso_8601_datetime).replace(tzinfo=timezone.utc)
```

Comment:
Don't we want to use `dateutil.parser.isoparse` (from `dateutil` package) here which is a more fully featured ISO-8601 format parser and would give us some more flexibility in the future?